### DISCUSSION TITLE
Signature verification on self-hosted keygen.sh

### DISCUSSION THREAD

**Discussion by hut36** [keygen-sh/community#96] [Issue]
I have problem creating public key when verifying signature, the code is(taken from [example-electron-license-activation](https://github.com/keygen-sh/example-electron-license-activation)):

```
 const verifyKey = crypto.createPublicKey({
    key: Buffer.from(KEYGEN_VERIFY_KEY, 'base64'),
    format: 'der',
    type: 'spki',
  })
```

KEYGEN_VERIFY_KEY  is from data.meta.keys.ed25519 of  GET /v1/accounts/{{account}}.

The error is:

```
Error: error:068000A8:asn1 encoding routines::wrong tag
    at Object.createPublicKey (node:internal/crypto/keys:619:12)
    ...
  opensslErrorStack: [ 'error:0688010A:asn1 encoding routines::nested asn1 error' ],
  library: 'asn1 encoding routines',
  reason: 'wrong tag',
  code: 'ERR_OSSL_ASN1_WRONG_TAG'
}
```

However, using a ed25519 key from official keygen.sh will not have error. I've no idea what I'm doing it wrong.

key doesn't work:
NTIzYzZkM2YwYTgwZjgyZGI0OGI3ZTliODBiMDA2Y2E3MGZmZGZkYWY2NTg0NzdkNGFlMzFlZDczOWNhOTQzNQ==

key works:
MCowBQYDK2VwAyEA6GAeSLaTg7pSAkX9B5cemD0G0ixCV8/YIwRgFHnO54g= (from [example-electron-license-activation](https://github.com/keygen-sh/example-electron-license-activation))

**Comment by ezekg** [comment]
That particular example only accepts a DER encoded public key, because that's what Node expects (it's an unfortunate choice by Node). Your Ed25519 key right now, via `meta.publicKeys.ed25519` is base64 encoded. So you will need to first decode it to get the hex encoded key, and then convert that to DER. Here's [a tutorial](https://keygen.sh/blog/how-to-use-hexadecimal-ed25519-keys-in-node/) on converting the hex encoded key to DER encoding.

**Comment by hut36** [comment]
Thanks, the Ed25519 OID is exactly what I was missing.
